generator/restriction_collector.cpp

```diff
@@ -0,0 +1,210 @@
+#include "generator/restriction_collector.hpp"
+
+#include "base/assert.hpp"
+#include "base/logging.hpp"
+#include "base/stl_helpers.hpp"
+#include "base/string_utils.hpp"
+
+#include "std/algorithm.hpp"
+#include "std/fstream.hpp"
+
+namespace
+{
+char const kNo[] = "No";
+char const kOnly[] = "Only";
+
+bool ParseLineOfNumbers(strings::SimpleTokenizer & iter, vector<uint64_t> & numbers)
+{
+  uint64_t number = 0;
+  while (iter)
+  {
+    if (!strings::to_uint64(*iter, number))
+      return false;
+    numbers.push_back(number);
+    ++iter;
+  }
+  return true;
+}
+}  // namespace
+
+namespace routing
+{
+RestrictionCollector::RestrictionCollector(string const & restrictionPath,
+                                           string const & featureId2OsmIdsPath)
+{
+  if (!ParseFeatureId2OsmIdsMapping(featureId2OsmIdsPath))
+    return;
+
+  if (!ParseRestrictions(restrictionPath))
+  {
+    m_restrictions.clear();
+    return;
+  }
+  ComposeRestrictions();
+  RemoveInvalidRestrictions();
+  LOG(LINFO, ("m_restrictions.size() =", m_restrictions.size()));
+}
+
+bool RestrictionCollector::IsValid() const
+{
+  return !m_restrictions.empty() &&
```

Comment:
I don't see a problem with empty m_restrictions list. Imagine an empty file with restrictions - there may be cases when a map doesn't have restrictions at all.